### ISSUE TITLE
Build fails on Raspbian Stretch

### ISSUE THREAD

**Issue by nlsdfnbch** [hypersad/atom-armv7l#19]
### Prerequisites

* [ X ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
Building on fresh Raspbian Stretch fails.

### Steps to Reproduce

1. Follow README.MD until `Building` section
2. clone repo as instructed and `cd` into repo dir
3. run `node script/build`

**Expected behavior:** [What you expect to happen]

Building finishes  without errors

**Actual behavior:** [What actually happens]
```
thanos@devBox:~/atom-armv7l $ node script/build
Node:   v6.10.2
Npm:    v3.10.10
Installing script dependencies

../src/onig-string.cc: In constructor ‘OnigString::OnigString(v8::Local<v8::String>)’:
../src/onig-string.cc:30:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   hasMultiByteChars = (value->Length() != utf8_length_);
                        ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
../deps/libgit2/deps/zlib/inflate.c: In function ‘inflateMark’:
../deps/libgit2/deps/zlib/inflate.c:1507:61: warning: left shift of negative value [-Wshift-negative-value]
     if (strm == Z_NULL || strm->state == Z_NULL) return -1L << 16;
                                                             ^~
../src/onig-string.cc: In constructor ‘OnigString::OnigString(v8::Local<v8::String>)’:
../src/onig-string.cc:30:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   hasMultiByteChars = (value->Length() != utf8_length_);
                        ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
../deps/libgit2/deps/zlib/inflate.c: In function ‘inflateMark’:
../deps/libgit2/deps/zlib/inflate.c:1507:61: warning: left shift of negative value [-Wshift-negative-value]
     if (strm == Z_NULL || strm->state == Z_NULL) return -1L << 16;

> oniguruma@6.1.0 install /home/thanos/atom-armv7l/node_modules/oniguruma
> node-gyp rebuild

make: Entering directory '/home/thanos/atom-armv7l/node_modules/oniguruma/build'
  CC(target) Release/obj.target/oniguruma/deps/onig/regcomp.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regenc.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regerror.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regexec.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regext.o
  CC(target) Release/obj.target/oniguruma/deps/onig/reggnu.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regparse.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regposerr.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regposix.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regsyntax.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regtrav.o
  CC(target) Release/obj.target/oniguruma/deps/onig/regversion.o
  CC(target) Release/obj.target/oniguruma/deps/onig/st.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/ascii.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/big5.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/cp1251.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/euc_jp.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/euc_kr.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/euc_tw.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/gb18030.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_1.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_2.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_3.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_4.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_5.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_6.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_7.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_8.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_9.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_10.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_11.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_13.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_14.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_15.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/iso8859_16.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/koi8.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/koi8_r.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/mktable.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/sjis.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/unicode.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/utf16_be.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/utf16_le.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/utf32_be.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/utf32_le.o
  CC(target) Release/obj.target/oniguruma/deps/onig/enc/utf8.o
  AR(target) Release/obj.target/oniguruma.a
  COPY Release/oniguruma.a
  CXX(target) Release/obj.target/onig_scanner/src/onig-result.o
  CXX(target) Release/obj.target/onig_scanner/src/onig-reg-exp.o
  CXX(target) Release/obj.target/onig_scanner/src/onig-scanner.o
  CXX(target) Release/obj.target/onig_scanner/src/onig-scanner-worker.o
  CXX(target) Release/obj.target/onig_scanner/src/onig-searcher.o
  CXX(target) Release/obj.target/onig_scanner/src/onig-string.o
  SOLINK_MODULE(target) Release/obj.target/onig_scanner.node
  COPY Release/onig_scanner.node
make: Leaving directory '/home/thanos/atom-armv7l/node_modules/oniguruma/build'

> runas@3.1.1 install /home/thanos/atom-armv7l/node_modules/runas
> node-gyp rebuild

make: Entering directory '/home/thanos/atom-armv7l/node_modules/runas/build'
  CXX(target) Release/obj.target/runas/src/main.o
  CXX(target) Release/obj.target/runas/src/runas_posix.o
  CXX(target) Release/obj.target/runas/src/fork.o
  SOLINK_MODULE(target) Release/obj.target/runas.node
  COPY Release/runas.node
make: Leaving directory '/home/thanos/atom-armv7l/node_modules/runas/build'


> superstring@1.1.0 install /home/thanos/atom-armv7l/node_modules/superstring
> node-gyp rebuild

make: Entering directory '/home/thanos/atom-armv7l/node_modules/superstring/build'
  CXX(target) Release/obj.target/superstring_core/src/core/patch.o
  CXX(target) Release/obj.target/superstring_core/src/core/point.o
  CXX(target) Release/obj.target/superstring_core/src/core/text.o
  CXX(target) Release/obj.target/superstring_core/src/core/marker-index.o
  CXX(target) Release/obj.target/superstring_core/src/core/buffer-offset-index.o
  AR(target) Release/obj.target/superstring_core.a
  COPY Release/superstring_core.a
  CXX(target) Release/obj.target/superstring/src/bindings/bindings.o
  CXX(target) Release/obj.target/superstring/src/bindings/marker-index-wrapper.o
  CXX(target) Release/obj.target/superstring/src/bindings/patch-wrapper.o
  CXX(target) Release/obj.target/superstring/src/bindings/point-wrapper.o
  CXX(target) Release/obj.target/superstring/src/bindings/buffer-offset-index-wrapper.o
  SOLINK_MODULE(target) Release/obj.target/superstring.node
  COPY Release/superstring.node
make: Leaving directory '/home/thanos/atom-armv7l/node_modules/superstring/build'

> keyboard-layout@2.0.11 install /home/thanos/atom-armv7l/node_modules/keyboard-layout
> node-gyp rebuild

npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.4.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN prefer global typescript@1.4.1 should be installed with -g
npm WARN prefer global jasmine-node@1.10.2 should be installed with -g
../src/onig-string.cc: In constructor ‘OnigString::OnigString(v8::Local<v8::String>)’:
../src/onig-string.cc:25:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   hasMultiByteChars = (value->Length() != utf8_length_);

child_process.js:504
    throw err;
    ^

Error: Command failed: npm --loglevel=error install



    at checkExecSyncError (child_process.js:481:13)
    at Object.execFileSync (child_process.js:501:13)
    at module.exports (/home/thanos/atom-armv7l/script/lib/install-script-dependencies.js:10:16)
    at Object.<anonymous> (/home/thanos/atom-armv7l/script/bootstrap:26:1)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
```
**Reproduces how often:** [What percentage of the time does it reproduce?]

### Additional Information

As stated, this was on a fresh install of Raspbian Stretch on a RaspberryPi2. Builds fine on RaspberryPi2 running Raspbian Jessie.

Let me know if this is something you may have to fix - I dont have any experience with NodeJS, hence I have no clue if this is something easily fixed or not.

**Comment by nlsdfnbch** [comment]
I've looked into npm a bit more and believe I understand it a bit better. I have tried manually installing the packages `minimatch`, ``graceful-fs` and `pug` (instead of `jade`, as suggested by the warning `npm` emits during build). However, `runas` kept not being installed, despite wiping `node_modules/` as instructed. 

I repeated the process (wiping `node_modules/`, installing / updating previously mentioned packages), with the addition of manually running `npm install runas` as well, but *still* it fails - insisting that `../build/Release/runas.node` cannot be found. 

Shouldn't it be detected now? I run all `npm` commands in the repo dir.

**Comment by daylightbrightledlight** [comment]
Hmm... I found this (may not be very useful): https://github.com/atom/atom/issues/3918

**Comment by nlsdfnbch** [comment]
Thanks @daylightbrightledlight !
It didn't fix the problem but it pointed me in the right direction. I manually installed all dependencies within the `script/` folder, instead of the base `atom-armv7l` directory. That fixed the above errors to an extend - the specified dependencies are not met.

It now appears that a module named `lodash` is missing. I've added the output of `script/build` as an attachment. I'll continue to look into this. Although I'm quite positive this isn't `stretch` related by now.
